### ISSUE TITLE
Assertion failed: [!context.is_nil()] we never got our context back

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#5740]
Got this on `v2.3.x` when running the table fuzzer:
```C++
error: Error in src/arch/runtime/context_switching.cc at line 122:
error: Assertion failed: [!context.is_nil()] we never got our context back
error: Backtrace:
Disconnected from server "e" c7c01477-5061-4da2-ac9a-8fce8079f710
error: Mon May  2 18:06:06 2016
       
       1 [0x2501908]: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:109
       2 [0x1c9f285]: backtrace_t::backtrace_t() at backtrace.cc:210 (discriminator 1)
       3 [0x1c9ffb9]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:331
       4 [0x1c9f1cf]: format_backtrace(bool) at backtrace.cc:205
       5 [0x24f8c35]: report_fatal_error(char const*, int, char const*, ...) at errors.cc:85 (discriminator 1)
       6 [0x1b1e40e]: artificial_stack_t::~artificial_stack_t() at context_switching.cc:122 (discriminator 1)
       7 [0x1b26929]: coro_t::~coro_t() at coroutines.cc:203
       8 [0x1b2698a]: coro_t::~coro_t() at coroutines.cc:211
       9 [0x1b2684e]: coro_t::maybe_evict_from_free_list() at coroutines.cc:196 (discriminator 1)
       10 [0x1b26a6e]: coro_t::switch_to_coro_with_protection(artificial_stack_context_ref_t*) at coroutines.cc:242
       11 [0x1b26d83]: coro_t::run() at coroutines.cc:314
       12 [0x1fbee30]: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const&, bool, auto_drainer_t::lock_t)> (table_query_client_t*, std::pair<peer_id_t, uuid_u>, table_query_bcard_t, bool, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const&, bool, auto_drainer_t::lock_t)> (table_query_client_t*, std::pair<peer_id_t, uuid_u>, table_query_bcard_t, bool, auto_drainer_t::lock_t)>&&) at coroutines.hpp:68
       13 [0x1fb984a]: table_query_client_t::update_registrant(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) at table_query_client.cc:493 (discriminator 1)
       14 [0x1fcc0d5]: void std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)>::operator()<std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, void>(table_query_client_t*, std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*&&) const at functional:601 (discriminator 3)
       15 [0x1fc9106]: void std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> (table_query_client_t*, std::_Placeholder<1>, std::_Placeholder<2>)>::__call<void, std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*&&, 0ul, 1ul, 2ul>(std::tuple<std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*&&>&&, std::_Index_tuple<0ul, 1ul, 2ul>) at functional:1297
       16 [0x1fc3e8a]: void std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> (table_query_client_t*, std::_Placeholder<1>, std::_Placeholder<2>)>::operator()<std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, void>(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*&&) at functional:1356
       17 [0x1fbfb6f]: std::_Function_handler<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*), std::_Bind<std::_Mem_fn<void (table_query_client_t::*)(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> (table_query_client_t*, std::_Placeholder<1>, std::_Placeholder<2>)> >::_M_invoke(std::_Any_data const&, std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) at functional:2073
       18 [0x1d3eefd]: std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)>::operator()(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) const at functional:2472
       19 [0x1d3d073]: watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&)#1}::operator()(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&) const at watchable_map.tcc:83
       20 [0x1d3ef6d]: void publisher_controller_t<std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> >::publish<watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&)#1}>(watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&)#1} const&) at pubsub.hpp:134
       21 [0x1d3d0f5]: watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*) at watchable_map.tcc:84 (discriminator 1)
       22 [0x1d3d7f1]: watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::set_key_no_equals(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const&) at watchable_map.tcc:256 (discriminator 1)
       23 [0x1d3b8eb]: cross_thread_watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::deliver(signal_t*) at cross_thread_watchable.tcc:75
       24 [0x1d3a5d2]: cross_thread_watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::cross_thread_watchable_map_var_t(watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>*, threadnum_t)::{lambda(signal_t*)#1}::operator()(signal_t*) const at cross_thread_watchable.tcc:48
       25 [0x1d3dad5]: std::_Function_handler<void (signal_t*), cross_thread_watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::cross_thread_watchable_map_var_t(watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>*, threadnum_t)::{lambda(signal_t*)#1}>::_M_invoke(std::_Any_data const&, signal_t*) at functional:2073
       26 [0x1bbcd5d]: std::function<void (signal_t*)>::operator()(signal_t*) const at functional:2472
       27 [0x1bbc8cd]: pump_coro_t::run(auto_drainer_t::lock_t) at pump_coro.cc:94
       28 [0x1bbdfeb]: void std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)>::operator()<auto_drainer_t::lock_t&, void>(pump_coro_t*, auto_drainer_t::lock_t&) const at functional:601 (discriminator 4)
       29 [0x1bbdf25]: void std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1297
       30 [0x1bbde5a]: void std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)>::operator()<, void>() at functional:1356
       31 [0x1bbde30]: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)> >::run_action() at callable_action.hpp:32
       32 [0x1b23dcc]: callable_action_wrapper_t::run() at runtime_utils.cc:48
       33 [0x1b26c49]: coro_t::run() at coroutines.cc:278
       34 [0x1bbcc0a]: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (pump_coro_t::*)(auto_drainer_t::lock_t)> (pump_coro_t*, auto_drainer_t::lock_t)>&&) at coroutines.hpp:68
       35 [0x1bbc461]: pump_coro_t::notify() at pump_coro.cc:31 (discriminator 1)
       36 [0x1d3ba75]: cross_thread_watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::on_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) at cross_thread_watchable.tcc:64
       37 [0x1d3a602]: cross_thread_watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::cross_thread_watchable_map_var_t(watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>*, threadnum_t)::{lambda(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)#2}::operator()(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) const at cross_thread_watchable.tcc:52
       38 [0x1d3dc10]: std::_Function_handler<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*), cross_thread_watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::cross_thread_watchable_map_var_t(watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>*, threadnum_t)::{lambda(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)#2}>::_M_invoke(std::_Any_data const&, std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) at functional:2073
       39 [0x1d3eefd]: std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)>::operator()(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*) const at functional:2472
       40 [0x1d3d073]: watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&)#1}::operator()(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&) const at watchable_map.tcc:83
       41 [0x1d3ef6d]: void publisher_controller_t<std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> >::publish<watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&)#1}>(watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*)> const&)#1} const&) at pubsub.hpp:134
       42 [0x1d3d0f5]: watchable_map_t<std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*) at watchable_map.tcc:84 (discriminator 1)
       43 [0x1d3b3a6]: watchable_map_transform_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t, std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::do_notify_change(std::pair<peer_id_t, uuid_u> const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*) at watchable_transform.hpp:55
       44 [0x1d3a366]: watchable_map_transform_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t, std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::watchable_map_transform_t(watchable_map_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t>*)::{lambda(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*)#1}::operator()(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*) const at watchable_transform.tcc:21
       45 [0x1d3d269]: std::_Function_handler<void (std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*), watchable_map_transform_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t, std::pair<peer_id_t, uuid_u>, table_query_bcard_t>::watchable_map_transform_t(watchable_map_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t>*)::{lambda(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*)#1}>::_M_invoke(std::_Any_data const&, std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*) at functional:2073
       46 [0x1c0377d]: std::function<void (std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*)>::operator()(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*) const at functional:2472
       47 [0x1c03169]: watchable_map_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t>::notify_change(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*)> const&)#1}::operator()(std::function<void (std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*)> const&) const at watchable_map.tcc:83
       48 [0x1c09601]: void publisher_controller_t<std::function<void (std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*)> >::publish<watchable_map_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t>::notify_change(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*)> const&)#1}>(watchable_map_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t>::notify_change(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*)> const&)#1} const&) at pubsub.hpp:134
       49 [0x1c031eb]: watchable_map_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t>::notify_change(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&, table_query_bcard_t const*, rwi_lock_assertion_t::write_acq_t*) at watchable_map.tcc:84 (discriminator 1)
       50 [0x1bfe59c]: watchable_map_var_t<std::pair<peer_id_t, std::pair<uuid_u, uuid_u> >, table_query_bcard_t>::delete_key(std::pair<peer_id_t, std::pair<uuid_u, uuid_u> > const&) at watchable_map.tcc:263 (discriminator 1)
       51 [0x1bfa59d]: directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>::do_update(peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&) at map_read_manager.tcc:93
       52 [0x1c1a1a1]: boost::_mfi::mf6<void, directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&>::operator()(directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>*, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&) const at mem_fn_template.hpp:732 (discriminator 5)
       53 [0x1c192fe]: void boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<std::pair<uuid_u, uuid_u> >, boost::_bi::value<boost::optional<table_query_bcard_t> > >::operator()<boost::_mfi::mf6<void, directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&>, boost::_bi::list0>(boost::_bi::type<void>, boost::_mfi::mf6<void, directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&>&, boost::_bi::list0&, int) at bind.hpp:670 (discriminator 1)
       54 [0x1c18d8d]: boost::_bi::bind_t<void, boost::_mfi::mf6<void, directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&>, boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<std::pair<uuid_u, uuid_u> >, boost::_bi::value<boost::optional<table_query_bcard_t> > > >::operator()() at bind_template.hpp:21
       55 [0x1c1864e]: callable_action_instance_t<boost::_bi::bind_t<void, boost::_mfi::mf6<void, directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&>, boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<std::pair<uuid_u, uuid_u> >, boost::_bi::value<boost::optional<table_query_bcard_t> > > > >::run_action() at callable_action.hpp:32
       56 [0x1b23dcc]: callable_action_wrapper_t::run() at runtime_utils.cc:48
       57 [0x1b26c49]: coro_t::run() at coroutines.cc:278
       58 [0x1bfe1df]: coro_t* coro_t::spawn_sometime<boost::_bi::bind_t<void, boost::_mfi::mf6<void, directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&>, boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<std::pair<uuid_u, uuid_u> >, boost::_bi::value<boost::optional<table_query_bcard_t> > > > >(boost::_bi::bind_t<void, boost::_mfi::mf6<void, directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, std::pair<uuid_u, uuid_u> const&, boost::optional<table_query_bcard_t> const&>, boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<std::pair<uuid_u, uuid_u> >, boost::_bi::value<boost::optional<table_query_bcard_t> > > >&&) at coroutines.hpp:68
       59 [0x1bf9e57]: directory_map_read_manager_t<std::pair<uuid_u, uuid_u>, table_query_bcard_t>::on_message(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, read_stream_t*) at map_read_manager.tcc:53
       60 [0x1bd62b5]: connectivity_cluster_t::run_t::handle(keepalive_tcp_conn_stream_t*, boost::optional<peer_id_t>, boost::optional<peer_address_t>, auto_drainer_t::lock_t, bool*, int) at cluster.cc:1317 (discriminator 1)
       61 [0x1bd3801]: connectivity_cluster_t::run_t::on_new_connection(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, int, auto_drainer_t::lock_t) at cluster.cc:326 (discriminator 3)
       62 [0x1bec6e2]: void std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, int, auto_drainer_t::lock_t)>::operator()<scoped_ptr_t<linux_tcp_conn_descriptor_t>&, int&, auto_drainer_t::lock_t&, void>(connectivity_cluster_t::run_t*, scoped_ptr_t<linux_tcp_conn_descriptor_t>&, int&, auto_drainer_t::lock_t&) const at functional:601 (discriminator 4)
       63 [0x1be9e48]: void std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, int, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, int, auto_drainer_t::lock_t)>::__call<void, scoped_ptr_t<linux_tcp_conn_descriptor_t>&, 0ul, 1ul, 2ul, 3ul>(std::tuple<scoped_ptr_t<linux_tcp_conn_descriptor_t>&>&&, std::_Index_tuple<0ul, 1ul, 2ul, 3ul>) at functional:1297
       64 [0x1be5d13]: void std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, int, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, int, auto_drainer_t::lock_t)>::operator()<scoped_ptr_t<linux_tcp_conn_descriptor_t>&, void>(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:1356
       65 [0x1be07b1]: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, int, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, int, auto_drainer_t::lock_t)> >::_M_invoke(std::_Any_data const&, scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:2073
       66 [0x1b10f8f]: std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)>::operator()(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) const at functional:2472
       67 [0x1b0d2dc]: linux_nonthrowing_tcp_listener_t::handle(int) at network.cc:1344 (discriminator 1)
       68 [0x1b17b3c]: void std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)>::operator()<int&, void>(linux_nonthrowing_tcp_listener_t*, int&) const at functional:601 (discriminator 3)
       69 [0x1b1777f]: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1297
       70 [0x1b17426]: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::operator()<, void>() at functional:1356
       71 [0x1b17302]: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >::run_action() at callable_action.hpp:32
       72 [0x1b23dcc]: callable_action_wrapper_t::run() at runtime_utils.cc:48
       73 [0x1b26c49]: coro_t::run() at coroutines.cc:278
       74 [0x1b10ef1]: void coro_t::spawn_now_dangerously<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >(std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>&&) at coroutines.hpp:62
       75 [0x1b0d10f]: linux_nonthrowing_tcp_listener_t::accept_loop(auto_drainer_t::lock_t) at network.cc:1312
error: Exiting.
```

**Comment by danielmewes** [comment]
I was running `table_fuzzer.test --servers 5 --kill --duration 900 --changefeeds`.
It might be a regression since 2.3.1.

**Comment by danielmewes** [comment]
A fix for this is in CR 3621 (this was indeed a regression which we hadn't released yet)

**Comment by danielmewes** [comment]
Fixed in v2.3.x a2e1481 and next 8775641.
